### PR TITLE
fix(typescript_indexer): emit refs to TYPE nodes in export statements

### PR DESCRIPTION
**Pull request by nbeloglazov**
Some symbols like interfaces don't have VALUE nodes and only have TYPE
nodes. This commit fixes refs for such symbols.

Tested:
  unit test

### CODE REVIEW

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -1241,14 +1241,25 @@ class Visitor {
           console.error(`TODO: export ${exp.name} has no symbol`);
           continue;
         }
-        // TODO: import a type, not just a value.
         const remoteSym = this.typeChecker.getAliasedSymbol(localSym);
+        const anchor = this.newAnchor(exp.name);
         const kExport = this.host.getSymbolName(remoteSym, TSNamespace.VALUE);
```
Comment (evmar) [inline kythe/typescript/indexer.ts]: It's not guaranteed that the export has a value.  For example, interface and type alias.

You can check:
`(flags & ts.SymbolFlags.Value) !== 0`

Be careful to check on the actual symbol, and not the alias of it.

File: kythe/typescript/indexer.ts [hunk 2]
```diff
@@ -1241,14 +1241,25 @@ class Visitor {
...
-        this.emitEdge(this.newAnchor(exp.name), EdgeKind.REF, kExport);
+        const kExportType =
+            this.host.getSymbolName(remoteSym, TSNamespace.TYPE);
+        this.emitEdge(anchor, EdgeKind.REF, kExport);
+        const flags =remoteSym.flags;
+        const isAlsoType = (flags & ts.SymbolFlags.Interface) ||
+            (flags & ts.SymbolFlags.Class) || (flags & ts.SymbolFlags.Enum);
```
Comment (evmar) [inline kythe/typescript/indexer.ts]: I think this misses some cases of types, such as type aliases.

I think you want just
`(flags & ts.SymbolFlags.Type) !== 0`

Comment (nbeloglazov) [reply kythe/typescript/indexer.ts]: Thanks! That's much better. Also added reexporting of type to the test.

**Comment by nbeloglazov** [comment]
@evmar can you take another look?

**Pull request merged**